lib/srv/desktop/windows_server.go

```diff
@@ -788,9 +786,13 @@ func (s *WindowsService) connectRDP(ctx context.Context, log logrus.FieldLogger,
 	var windowsUser string
 	authorize := func(login string) error {
 		windowsUser = login // capture attempted login user
+		mfaParams := services.AccessMFAParams{
+			Verified:       identity.MFAVerified != "",
+			AlwaysRequired: authPref.GetRequireSessionMFA(),
```

Comment:
The only implementation for this my editor finds is [here](https://github.com/gravitational/teleport/blob/a943a5f09dcd0417eff52221d0485013a818ca13/api/types/authentication.go#L345-L349), which returns the field corresponding to this definition [here](https://github.com/gravitational/teleport/blob/a943a5f09dcd0417eff52221d0485013a818ca13/api/types/types.proto#L993-L995), which (if I'm not mistaken) corresponds to the [cluster-wide per-session MFA setting](https://goteleport.com/docs/access-controls/guides/per-session-mfa/#cluster-wide). However according to that documentation, we also support [per-role mfa](https://goteleport.com/docs/access-controls/guides/per-session-mfa/#per-role) which this doesn't seem to account for.

Reply:
I think this is fine. `AlwaysRequired` means is enabled at the cluster level, so there's no need to check the roles.

If you look at `CheckAccess()` in lib/services/role.go, you'll see we also consult the roles if `AlwaysRequired` was false:

```
// if MFA is not verified and we require session MFA, deny access
if role.GetOptions().RequireSessionMFA {
```